### PR TITLE
Change a byte[] in KnownColorTable to be ReadOnlySpan<byte>

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @safern

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Review by safern — APPROVED** [review]
Curious did you find this in auditing memory on some app or anything? 

Should we also update `s_colorValueTable` to `ReadOnlySpan<uint>` or that doesn't help?

File: src/libraries/Common/src/System/Drawing/KnownColorTable.cs [hunk 1]
```diff
@@ -238,7 +238,7 @@ internal static class KnownColorTable
         };
 
         // All known color kinds (in order of definition in the KnownColor enum).
-        public static readonly byte[] s_colorKindTable = new byte[]
+        public static ReadOnlySpan<byte> ColorKindTable => new byte[]
```
[the remark this answers is not in the document]
Comment (Tornhoof) [reply src/libraries/Common/src/System/Drawing/KnownColorTable.cs:241]: It's an expression-bodied property now (see the ``=>``)

**Comment by gfoidl** [comment]
> Should we also update `s_colorValueTable` to `ReadOnlySpan<uint>` or that doesn't help?

This doesn't help. The C# compiler's optimization to refer to the data-segment of the assembly directly only works for (primitive) types with one byte size (due to endianess concerns). See e.g. [this sharplab](https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCjYG9aN4gDgE4CWAbgIYAuApogErXkAmA8gHYA2AngMr7lsAPACMuNAHyIAIlXIAGRAF5JbagHdEomgG0AuokyI5ADwiIAvgG4ceImSq0Gzdtz4DBAZy3VJMyuTNlRFUNLzFqPQMjUwtrGDwCEgoaekZWTl5+IWEAexyOX1kwJRV1TTyOSMMAM3IOD1orBIB6ZsRqjmJ8REocxABySkIAV2p+xAEmHoALWiZZRGIPREJqAFphhqnqwhyAW01qapzVm1w7ZMc0l0z3YeI2SkL/JCCQxHvHqsQzKzPE+wpJzpVxZQTsagAIXCPCGwwAxk9pLIACwlYJlCHQmiwkaI74mX5xcywWAIRAeOGIxBYmFUyhYf7k7yIABqdVGcQSTKQIAgADZFnt8J14cRKIgcvhqIQqCcaapsdRcQjKAAKFkUDijACU6PetJx9Ki7O1tEUiC1oz+MHMQA==).

**Comment by stephentoub** [comment]
We'll need https://github.com/dotnet/runtime/issues/24961 and Roslyn support for it before we can start using this with types other than byte, sbyte, and bool.

**Comment by stephentoub** [comment]
> Curious did you find this in auditing memory on some app or anything?

I was going back through some PRs I hadn't reviewed and saw the array rather than span being introduced in https://github.com/dotnet/runtime/pull/50489.

**Pull request merged**